src/python/pants/option/config.py

```diff
@@ -390,10 +390,10 @@ def get_value(self, section: str, option: str) -> Optional[str]:
     section_values = self._find_section_values(section)
     if section_values is None:
       raise configparser.NoSectionError(section)
-    if option in self.defaults:
-      return self._stringify_val(self.defaults[option])
     if option not in section_values:
-      raise configparser.NoOptionError(option, section)
+      if option not in self.defaults:
+        raise configparser.NoOptionError(option, section)
+      return self._stringify_val(self.defaults[option])
```

Comment:
These changes were necessary to get it working when an option overrides a DEFAULT option with the same name.